MobiFlight/Joysticks/VKB/VKBDevice.cs

```diff
@@ -0,0 +1,192 @@
+﻿using HidSharp;
+using HidSharp.Reports;
+using HidSharp.Reports.Input;
+using System;
+using System.Collections.Generic;
+
+namespace MobiFlight.Joysticks.VKB
+{
+    internal class VKBDevice : Joystick
+    {
+        private HidStream Stream;
+        private readonly HidDevice Device;
+        private readonly JoystickDefinition Definition;
+        private readonly VKBLedContainer Lights = new VKBLedContainer();
+        private HidDeviceInputReceiver InputReceiver;
+        private readonly byte[] InputReportBuffer = new byte[64];
+        private readonly SortedList<byte, VKBEncoder> Encoders = new SortedList<byte, VKBEncoder>();
+        int lastSeqNo = -1;
+
+        public VKBDevice(SharpDX.DirectInput.Joystick joystick, JoystickDefinition definition) : base(joystick, definition)
+        {
+            Definition = definition;
+            if (Device == null)
+            {
+                Device = GetMatchingHidDevice(joystick);
+                if (Device == null) return;
+            }
+        }
+        public override void Connect(IntPtr handle)
+        {
+            base.Connect(handle);
+            if (Stream == null)
+            {
+                Stream = Device.Open();
+                Stream.ReadTimeout = System.Threading.Timeout.Infinite;
+            }
+
+            if (InputReceiver == null)
+            {
+                InputReceiver = new ReportDescriptor(VKBHidReport.Descriptor).CreateHidDeviceInputReceiver();
+                // We use our own descriptor since the one we get from Windows/HIDSharp has been altered and encoder data is missing
+                InputReceiver.Received += OnHidReportReceived;
+                InputReceiver.Start(Stream);
+            }
+        }
+        protected override void SendData(byte[] data)
+        {
+            // VKBLedContainer has its own handling that replaces RequiresOutputUpdate, so we just send.
+            // Don't try and send data if no outputs are defined.
+            if (Definition?.Outputs == null || Definition?.Outputs.Count == 0)
+            {
+                return;
+            }
+            Stream?.SetFeature(data);
+
+        }
+        protected override void EnumerateDevices()
+        {
+            base.EnumerateDevices();
+            SortedList<byte, JoystickDevice> EncoderDecList = new SortedList<byte, JoystickDevice>();
+            SortedList<byte, JoystickDevice> EncoderIncList = new SortedList<byte, JoystickDevice>();
```

Comment:
This should be `var` EncoderDecList/EncoderIncList to stay consistent with our (ongoing!) style update to use var where possible. Also applies to all other variable definitions in this PR, where possible use `var`.